src/lease/mod.rs

```diff
@@ -150,7 +89,7 @@ impl LeaseKeepAliveTunnel {
 impl Shutdown for LeaseKeepAliveTunnel {
     async fn shutdown(&mut self) -> Result<()> {
         if let Some(shutdown) = self.shutdown.take() {
-            shutdown.send(()).map_err(|_| Error::ChannelClosed)?;
+            shutdown.send(()).await.map_err(|_| Error::ChannelClosed)?;
```

Comment:
this kind of error conversion lost the original error context, use this_error to auto convert error by using ?

Reply:
Synced offline. Will address this later.